controllers/update.go

```diff
@@ -84,6 +85,19 @@ func (r *DevfileRegistryReconciler) updateDeployment(ctx context.Context, cr *re
 			needsUpdating = true
 		}
 	}
+	headlessStatus := dep.Spec.Template.Spec.Containers[0].Env
+
+	for _, env := range headlessStatus {
+		if env.Name == "REGISTRY_HEADLESS" {
```

Comment:
I have some time to check this part of the code, so I'm not sure I understand the purpose of the env var usage here. Could you elaborate?

From my point of view, I would expect us here to:

* Compare the existing headless status with the new one.
* We could fetch the existing from `dep.Spec.Template.Spec.Headless`
* We could create (IIRC we don't have one) a function to get the headless status from the `registryv1alpha1.DevfileRegistry` resource.
* In case the two of them are different assign the current value to the existing one and update the `needsUpdating` variable to `true`.

Reply:
`We could create (IIRC we don't have one) a function to get the headless status from the registryv1alpha1.DevfileRegistry resource.`
im comparing on the basis of deployment , i also taught about that but saw this 
https://github.com/Horiodino/registry-operator/blob/headless-boolean/pkg/registry/deployment.go#L286
so i didn't added anything.

is that same thing or different ?

Reply:
That looks like it is setting the headless value but I believe @thepetk is wanting a checker function to easily grab the headless value from a CR

Reply:
@Horiodino yes `dep` is a CR but the built-in Kubernetes Deployment `CR` to be more specific.

Reply:
ah its not the user defined one right , theirs not any wrapper for getting that CR  correct ?

Reply:
Not that I can think of off the top of my head, I think just abstracting your piece of code into a function that takes in a deployment cr and fetches the headless env and returns what it is would be good to add pieces that can easily be tested with unit tests. So I would recommend doing that and adding unit tests for it. Side note, you can deploy the operator and registry on Kubernetes (Minikube/Kind) locally and can verify what happens when you change the headless value, I know during my run throughs it wasn't adding/removing the registry-viewer container as expected